modules/preview-react/radio/lib/RadioInput.tsx

```diff
@@ -0,0 +1,218 @@
+  createElemPropsHook,
+} from '@workday/canvas-kit-react/common';
+import {colors, inputColors, spaceNumbers, borderRadius} from '@workday/canvas-kit-react/tokens';
+import {useRadioModel} from './hooks/useRadioModel';
+import {RadioLabelProps, RadioLabelContext} from './RadioLabel';
+import {Box, Flex} from '@workday/canvas-kit-react/layout';
+
+const radioWidth = 18;
+const rippleRadius = (spaceNumbers.l - radioWidth) / 2;
+const radioHeight = 18;
+
+const StyledRadioInput = styled(Box.as('input'))<RadioLabelProps & StyledType>(
+  {
+    '&:focus, &:active': {
+      outline: 'transparent',
+    },
+  },
+  ({
+    disabled,
+    variant,
+    theme: {
+      canvas: {
+        palette: {
+          primary: themePrimary,
+          common: {focusOutline: themeFocusOutline},
+        },
+      },
+    },
+  }) => ({
+    cursor: disabled ? undefined : 'pointer',
+    opacity: disabled && variant === 'inverse' ? '.4' : '1',
+    height: '18px',
+    width: '18px',
+
+    // Circle element styles the radio as checked or unchecked
+    ':after': {
+      content: "''",
+      display: 'flex',
+      flexDirection: 'column',
+      alignItems: 'center',
+      backgroundColor: disabled ? inputColors.disabled.background : colors.frenchVanilla100,
+      borderRadius: '999px',
+      boxSizing: 'border-box',
+      border: `1px solid`,
+      borderColor: disabled
+        ? inputColors.disabled.border
+        : variant === 'inverse'
+        ? colors.soap300
+        : inputColors.border,
+      height: '18px',
+      width: '18px',
+      justifyContent: 'center',
+      pointerEvents: 'none',
+      position: 'absolute',
+      transition: 'border 200ms ease, background 200ms',
+      opacity: disabled && variant === 'inverse' ? '.4' : '1',
+    },
+
+    '&:hover:after': {
+      borderColor: disabled
+        ? inputColors.disabled.border
+        : variant === 'inverse'
+        ? colors.soap300
+        : colors.licorice500,
+    },
+
+    '&:focus:after': {
+      borderColor: disabled
+        ? inputColors.disabled.border
+        : variant === 'inverse'
+        ? colors.soap300
+        : colors.blueberry400,
+    },
+
+    '&:checked:after': {
+      backgroundColor: variant === 'inverse' ? themePrimary.main : colors.frenchVanilla100,
+      border: `5px solid`,
+      borderColor: variant === 'inverse' ? colors.frenchVanilla100 : themePrimary.main,
+    },
+
+    '&:focus:after, &:focus:hover:after': {
+      outline: 'transparent',
+
+      ...focusRing({
+        width: variant === 'inverse' ? 2 : 1,
+        separation: 0,
+        animate: false,
+        innerColor: variant === 'inverse' ? colors.blackPepper400 : colors.frenchVanilla100,
+        outerColor: variant === 'inverse' ? colors.frenchVanilla100 : colors.blueberry400,
+      }),
+    },
+
+    '&:focus:checked:after, &:focus:hover:checked:after': {
+      outline: 'transparent',
+      ...focusRing({
+        width: variant === 'inverse' ? 2 : 2,
```

Comment:
Is this ternary correct? if yes, should it be `width: 2`?

Reply:
fixing